app/controllers/favorite_controller.rb

```diff
@@ -6,12 +6,19 @@ def index
     end
   end
 
-  def update
+  def create
     pet = Pet.find(params[:pet_id])
     favorite.add_pet(pet.id)
     session[:favorite] = @favorite.contents
     flash[:pet_fav] = "Pet added to favorites."
     redirect_to "/pets/#{pet.id}"
   end
 
+  def update
+    pet = Pet.find(params[:pet_id])
+    favorite.remove_pet(pet.id)
+    session[:favorite] = @favorite.contents
+    flash[:pet_fav] = "Pet removed from favorites."
```

Comment:
I think this will need to be a different error message title to display the proper message